guides/keras_hub/mcp_with_keras_hub.py

```diff
@@ -0,0 +1,828 @@
+
+    This function creates and registers our three main tools:
+    - Weather tool for getting weather information
+    - Calculator tool for mathematical computations
+    - Search tool for information retrieval
+
+    Args:
+        tool_registry: The MCPToolRegistry instance to register tools with
+    """
+    # Create and register the weather tool
+    weather_tool_instance = MCPTool(
+        name="weather", description="Get weather for a city", function=weather_tool
+    )
+    tool_registry.register_tool(weather_tool_instance)
+
+    # Create and register the calculator tool
+    calculator_tool_instance = MCPTool(
+        name="calculator",
+        description="Calculate math expressions",
+        function=calculator_tool,
+    )
+    tool_registry.register_tool(calculator_tool_instance)
+
+    # Create and register the search tool
+    search_tool_instance = MCPTool(
+        name="search", description="Search for information", function=search_tool
+    )
+    tool_registry.register_tool(search_tool_instance)
+
+
+"""
+## Complete MCP Demonstration
+
+This function orchestrates the entire MCP system demonstration:
+
+1. **Sets up the tool registry** - Registers all available tools (weather, calculator, search)
+2. **Loads the AI model** - Gets the Gemma3 Instruct 1B model ready
+3. **Creates the MCP client** - Connects everything together
+4. **Runs demonstration examples** - Shows weather, calculator, and search in action
+5. **Demonstrates the system** - Proves MCP works with real tool execution
+
+Think of this as the grand finale where all the components come together to create something amazing!
+"""
+
+
+def main():
+    print("🎯 Simple MCP with KerasHub - Working Implementation")
+    print("=" * 70)
+
+    # Set up our tool registry
+    tool_registry = MCPToolRegistry()
+    _register_tools(tool_registry)
+
+    # Load our AI model
+    model = _load_model()
+
+    # Create our MCP client
+    client = MCPClient(model, tool_registry)
+
+    print("🚀 Starting MCP demonstration...")
+    print("=" * 50)
+
+    # Example 1: Weather Information
+    print("Example 1: Weather Information")
+    print("=" * 50)
+    user_input = "What's the weather like in Tokyo?"
+    print(f"🤖 User: {user_input}")
+
+    response = client.chat(user_input)
+    print(f"💬 Response: {response}")
+    print()
+
+    # Example 2: Calculator
+    print("Example 2: Calculator")
+    print("=" * 50)
+    user_input = "Calculate 15 * 23 + 7"
+    print(f"🤖 User: {user_input}")
+
+    response = client.chat(user_input)
+    print(f"💬 Response: {response}")
+    print()
+
+    # Example 3: Search
+    print("Example 3: Search")
+    print("=" * 50)
+    user_input = "Search for information about machine learning"
+    print(f"🤖 User: {user_input}")
+
+    response = client.chat(user_input)
+    print(f"💬 Response: {response}")
+    print()
+
+    print("🎉 MCP demonstration completed successfully!")
+
+
+if __name__ == "__main__":
+    main()
+
```

Comment:
the simulated data is okay - not sure if this is possible but is there a real tool API we can actually show?